docs/source/model_design/programming_vivarium/debugging/index.rst

```diff
@@ -0,0 +1,190 @@
+..
+  Section title decorators for this document:
+  
+  ==============
+  Document Title
+  ==============
+  Section Level 1
+  ---------------
+  Section Level 2
+  +++++++++++++++
+  Section Level 3
+  ~~~~~~~~~~~~~~~
+  Section Level 4
+  ^^^^^^^^^^^^^^^
+  Section Level 5
+  '''''''''''''''
+
+  The depth of each section level is determined by the order in which each
+  decorator is encountered below. If you need an even deeper section level, just
+  choose a new decorator symbol from the list here:
+  https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections
+  And then add it to the list of decorators above.
+
+.. _debugging_rt:
+
+===============
+Basic Debugging
+===============
+
+.. contents::
+   :local:
+   :depth: 2
+
+Debugging is an art and science that we couldn't hope to cover on this 
+single page. Rather than attempt such foolishness, we we review a few simple 
+techniques in the context of a targeted example here. If you find the 
+bug to be outside the scope of this page, it's likely time to ask an 
+engineer for help.  
+
+Making Sense of the Stack Trace
+-------------------------------
+
+The stack trace is an output that the model provides to show what 
+error message was received and trace back the flow of code to what 
+caused the error. Often the immediate line of code that generated the error will be 
+significantly removed from the line of code that will need to be changed 
+to fix the error. 
+
+Let's work through an example to understand this. Say a line of code 
```

Comment:
Consider including an example of code that results in such a stack trace, so the reader can see exactly what you are referring to.

Reply:
I agree, I think having an actual code example would make this easier to follow.